minimesos/src/main/java/com/containersol/minimesos/container/AbstractContainer.java

```diff
@@ -127,6 +128,51 @@ public String getContainerId() {
         return containerId;
     }
 
+    @Override
+    public URI getServiceUrl() {
+
+        URI serviceUri = null;
+
+        String protocol = getServiceProtocol();
+        String host = getIpAddress();
+        int port = getServicePort();
+        String path = getServicePath();
+
+        if (StringUtils.isNotEmpty(host)) {
+            try {
+                serviceUri = new URI(protocol, null, host, port, path, null, null);
+            } catch (URISyntaxException e) {
+                throw new MinimesosException("Failed to form service URL for " + getName(), e);
+            }
+        }
+
+        return serviceUri;
+    }
+
+    /**
+     * Enables derived classes to override
+     * @return protocol of the service
+     */
+    protected String getServiceProtocol() {
+        return "http";
+    }
+
+    /**
+     * Enables derived classes to override
+     * @return port of the service
+     */
+    protected int getServicePort() {
+        return 80;
+    }
+
+    /**
+     * Enables derived classes to override
+     * @return protocol of the service
+     */
+    protected String getServicePath() {
+        return "";
+    }
+
```

Comment:
I though about this, but decided to provide default implementation. This somewhat limits the number of methods derived classes have to implement. For example, it implementation of these methods would be useless in Registrator